src/cirrus/builtins/functions/update-state/lambda_function.py

```diff
@@ -191,4 +192,4 @@ def lambda_handler(event, context={}):
         logger.info("Status does not support updates")
         return
 
-    status_update_map[status](payload, error)
+    status_update_map[status](input_payload, output_payload, error)
```

Comment:
in order to **not** force the function signature between different state handlers (failed, completed) to be identical, I'd remove this handler map and just go with 

pseudo code:
```python
if status == FAILED:
    workflow_failed(item_id=input_payload['id'], error=error)  # NOTE: only pass what's needed of a complex obj (id)
    
if status == SUCCEEDED:
     workflow_succeeded(item_id=input_payload['id'], output_payload=output_payload)  # NOTE: only pass what's needed of a complex obj (id)